### ISSUE TITLE
Problem with AutoRest.exe generated deserializer

### ISSUE THREAD

**Issue by devigned** [Azure/autorest#54]
The auto generated deserializer seems to have a problem with deserializing dictionary types.

The swagger definition has a property of: 
```json
myDictionary": { "type":"object", "additionalProperties":{"type":"string"} }
```

AutoRest produces an correct member property (public IDictionary<string, string> MyDictionary), but the serializer for that property is as the following:

```csharp
JToken stringDictionary = ((JToken)inputObject["string"]); 
if (stringDictionary != null && stringDictionary.Type != JTokenType.Null) 
{ 
  foreach (JProperty property in stringDictionary) 
  { 
    this.MyDictionary.Add(((string)property.Name), ((string)property.Value)); 
  }
}
```

It seems to work as expected if i change the deserializer to the following:

```csharp
JToken stringDictionary = ((JToken)inputObject["myDictionary"]); 
if (stringDictionary != null && stringDictionary.Type != JTokenType.Null) 
{ 
  foreach (JProperty property in stringDictionary) 
  { 
    this.MyDictionary.Add(((string)property.Name), ((string)property.Value)); 
  } 
}
```

**Comment by markcowl** [comment]
David, can you provide the actual swagger you were using?

**Comment by PeterEeg** [comment]
Hi, I was the one that reported the issue to David to start with.

I have been using Swachbuckle, following the getting started guide, to produce the swagger definition (https://github.com/domaindrivendev/Swashbuckle)

My REST endpoind exposes a model with a Dictionary:

```
	public class Product
	{
		public Dictionary<string, string> MyDictionary { get; set; }
		public int Id { get; set; }
	}
```

The generated swagger:

```
"swagger": "2.0",
	"info": {
		"version": "v1",
		"title": "WebApplication2"
	},
	"host": "localhost:56997",
	"schemes": ["http"],
	"paths": {
		"/api/Products": {
			"get": {
				"tags": ["Products"],
				"operationId": "Products_GetAllProducts",
				"consumes": [],
				"produces": ["application/json",
				"text/json",
				"application/xml",
				"text/xml"],
				"responses": {
					"200": {
						"description": "OK",
						"schema": {
							"type": "array",
							"items": {
								"$ref": "#/definitions/Product"
							}
						}
					}
				},
				"deprecated": false
			}
		},
		"/api/Products/{id}": {
			"get": {
				"tags": ["Products"],
				"operationId": "Products_GetProduct",
				"consumes": [],
				"produces": ["application/json",
				"text/json",
				"application/xml",
				"text/xml"],
				"parameters": [{
					"name": "id",
					"in": "path",
					"required": true,
					"type": "integer",
					"format": "int32"
				}],
				"responses": {
					"200": {
						"description": "OK",
						"schema": {
							"$ref": "#/definitions/Object"
						}
					}
				},
				"deprecated": false
			}
		}
	},
	"definitions": {
		"Product": {
			"type": "object",
			"properties": {
				"myDictionary": {
					"type": "object",
					"additionalProperties": {
						"type": "string"
					}
				},
				"id": {
					"format": "int32",
					"type": "integer"
				}
			}
		},
		"Object": {
			"type": "object",
			"properties": {
				
			}
		}
	}
}
```

**Comment by devigned** [comment]
We are looking into this. Thank you for the info for repro.

Also, beware of the following:
```json
"Object": {
            "type": "object",
            "properties": {
            }
        }
```
I will deserialize as a string.

**Comment by PeterEeg** [comment]
Great, thanks.

**Comment by hovsepm** [comment]
Fixed in Release 0.10.0 - [nuget](https://www.nuget.org/packages/AutoRest/), [chocolatey](https://chocolatey.org/packages/AutoRest), [github] (https://github.com/Azure/autorest/releases/tag/AutoRest-0.10.0)
